app/src/main/kotlin/com/hippo/ehviewer/ui/main/FabLayout.kt

```diff
@@ -83,25 +153,26 @@ fun FabLayout(
             },
             contentAlignment = Alignment.Center,
         ) {
-            val animatedProgress by animateFloatMergePredictiveBackAsState(
-                expanded,
-                tween(FAB_ANIMATE_TIME, if (delayOnExpand) FAB_ANIMATE_TIME else 0),
-                finishedListener = { delayOnHide = expanded },
-            ) {
-                onExpandChanged(false)
+            val animatedProgress by state.expandProgress.asState()
+            PredictiveBackHandler(updatedExpanded) { flow ->
+                try {
+                    state.mutatorMutex.mutate(MutatePriority.UserInput) {
+                        flow.collect {
+                            val eased = EaseOut.transform(it.progress)
+                            state.expandProgress.snapTo(1 - eased)
+                        }
+                    }
+                    onExpandChanged(false)
+                } catch (e: CancellationException) {
+                    state.expand()
+                }
             }
-            FloatingActionButton(
-                onClick = { onExpandChanged(!expanded) },
-            ) {
+            FloatingActionButton(onClick = { onExpandChanged(!updatedExpanded) }) {
                 Icon(
                     imageVector = Icons.Default.Close,
                     contentDescription = null,
-                    modifier = if (showAddButton) {
-                        Modifier.graphicsLayer {
-                            rotationZ = lerp(0f, -135f, animatedProgress)
-                        }
-                    } else {
-                        Modifier
+                    modifier = Modifier.graphicsLayer {
+                        rotationZ = lerp(-135f, 0f, animatedProgress)
```

Comment:
We don't need rotation in DownloadsScreen.

Reply:
It looks better
You mean let the primary and secondary appear together?

Reply:
In DownloadsScreen the primary FAB will never show without secondary FABs, so the on expand / collapse animation for the primary FAB is redundant.

Reply:
But this makes fab animation consistent with GalleryList/FavScreen one
And I do think this is more harmony

Reply:
It's different in GalleryList/FavScreen as primary FABs are always shown there.
In Downloads the primary FAB will first rotate 90°, then immediately rotate 135°, which I think it's too much for a sequence of animation.

Reply:
Primary Fab is not always here in GalleryList, it will hide when user scroll
Now we can introduce that in FavScreen as well

Reply:
I don't think this two-stage animation is bad
User will never complain that "anim is slow and affect experience"

Reply:
> It's different in GalleryList/FavScreen as primary FABs are always shown there. In Downloads the primary FAB will first rotate 90°, then immediately rotate 135°, which I think it's too much for a sequence of animation.

Not immediately, expand is happen after appear